### PR TITLE
[WIP] nlp-architect intent extraction using TFDataSet and TFOptimizer

### PR DESCRIPTION
**Pull request by hkvision**
Due to some error, only enable the intent extraction part of MultiTaskIntentModel with the following modifications / constraints:
- Use keras (2.1.6) instead of tf.keras
- Use zero-based label instead of one-hot especially for validation
- nlp-architect only supports Python 3.5+

### CODE REVIEW

**Comment by yangw1234** [comment]
#1012 has fixed those error. I tested the full model without metrics, it works fine

**Comment by hkvision** [comment]
@yangw1234 Wow! Great! Thanks!

**Comment by hkvision** [comment]
Close this PR first. Will open another one to update the code.
